src/TwigComponent/src/EventListener/PreRenderEvent.php

```diff
@@ -77,6 +77,11 @@ public function setVariables(array $variables): self
 
     public function getMetadata(): ComponentMetadata
     {
-        return $this->metadata;
+        return $this->mounted->getMetadata();
+    }
+
+    public function getMountedComponent(): MountedComponent
```

Comment:
As discussed with @weaverryan Slack, we'd like to keep the `MountedComponent` object "internal". This getter is required for `AddLiveAttributesSubscriber` but maybe we should mark it as `@internal`?